### ISSUE TITLE
loss issue: loss is 0 when RL training, and  the issue come from source code that author provided

### ISSUE THREAD

**Issue by HillDing** [Liuziyu77/Visual-RFT#122]
The loss always equal to 0 at the training process, as follow
loss值始终为0，如下图所示

![Image](https://github.com/user-attachments/assets/9a00bada-946a-4b83-baf7-65486908e8a4)

and I find the problem come from the 488 line of grpo_trainer.py
![Image](https://github.com/user-attachments/assets/61f8dc6f-d535-4f56-aa8f-bc6686fcb46a)
per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1)
per_token_logps mins per_token_logps always equal to 0, so how to fix it?

**Comment by corkiyao** [comment]
> The loss always equal to 0 at the training process, as follow loss值始终为0，如下图所示
> 
> ![Image](https://github.com/user-attachments/assets/9a00bada-946a-4b83-baf7-65486908e8a4)
> 
> and I find the problem come from the 488 line of grpo_trainer.py ![Image](https://github.com/user-attachments/assets/61f8dc6f-d535-4f56-aa8f-bc6686fcb46a) per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1) per_token_logps mins per_token_logps always equal to 0, so how to fix it?

你好，请问有什么思路吗？我也遇到这个问题了

**Comment by HillDing** [comment]
> > The loss always equal to 0 at the training process, as follow loss值始终为0，如下图所示
> > ![Image](https://github.com/user-attachments/assets/9a00bada-946a-4b83-baf7-65486908e8a4)
> > and I find the problem come from the 488 line of grpo_trainer.py ![Image](https://github.com/user-attachments/assets/61f8dc6f-d535-4f56-aa8f-bc6686fcb46a) per_token_loss = torch.exp(per_token_logps - per_token_logps.detach()) * advantages.unsqueeze(1) per_token_logps mins per_token_logps always equal to 0, so how to fix it?
> 
> 你好，请问有什么思路吗？我也遇到这个问题了

not yet, Im try to solve it but have no idea

**Comment by lyccnb** [comment]
我也遇到了这个情况。我发现，代码中设置的 generate_config 被模型的默认参数覆盖掉了，所以我直接修改了模型文件中的 generate_config.json
`{
    "bos_token_id": 151643,
    "pad_token_id": 151643,
    "do_sample": true,
    "eos_token_id": [
      151645,
      151643
    ],
    "repetition_penalty": 1.0,
    "temperature": 1.0,
    "top_p": 0.8,
    "top_k": 5,
    "transformers_version": "4.37.0"
  }`
然后就正常了。
我认为这是因为之前默认的 generate_config 中，temporature 为 0.01，top_p = 0.001 导致模型生成的 generations 都相同，进而计算出的 advantages 为 0，最后导致损失为 0。

**Comment by corkiyao** [comment]
> 我也遇到了这个情况。我发现，代码中设置的 generate_config 被模型的默认参数覆盖掉了，所以我直接修改了模型文件中的 generate_config.json `{ "bos_token_id": 151643, "pad_token_id": 151643, "do_sample": true, "eos_token_id": [ 151645, 151643 ], "repetition_penalty": 1.0, "temperature": 1.0, "top_p": 0.8, "top_k": 5, "transformers_version": "4.37.0" }` 然后就正常了。 我认为这是因为之前默认的 generate_config 中，temporature 为 0.01，top_p = 0.001 导致模型生成的 generations 都相同，进而计算出的 advantages 为 0，最后导致损失为 0。

我也尝试了在generation_config修改了像你一样的参数，但是损失值依旧保持不变，请问你后续使用其他思路解决了吗？

**Comment by LilaKen** [comment]
same wrong， but in qwen2-vl i don't meet this problem. in qwen2.5-vl with lora, i meet the same problem.

**Comment by Cathrii** [comment]
我也遇到了这个问题，请问你解决了吗

**Comment by HillDing** [comment]
> > 我也遇到了这个情况。我发现，代码中设置的 generate_config 被模型的默认参数覆盖掉了，所以我直接修改了模型文件中的 generate_config.json `{ "bos_token_id": 151643, "pad_token_id": 151643, "do_sample": true, "eos_token_id": [ 151645, 151643 ], "repetition_penalty": 1.0, "temperature": 1.0, "top_p": 0.8, "top_k": 5, "transformers_version": "4.37.0" }` 然后就正常了。 我认为这是因为之前默认的 generate_config 中，temporature 为 0.01，top_p = 0.001 导致模型生成的 generations 都相同，进而计算出的 advantages 为 0，最后导致损失为 0。
> 
> 我也尝试了在generation_config修改了像你一样的参数，但是损失值依旧保持不变，请问你后续使用其他思路解决了吗？

while I change the temperature from 0 to 1, the loss become normal value.
